File: screen_brightness_android/android/src/main/kotlin/com/aaassseee/screen_brightness_android/stream_handler/BaseStreamHandler.kt
package com.aaassseee.screen_brightness_android.stream_handler

import io.flutter.plugin.common.EventChannel

open class BaseStreamHandler: EventChannel.StreamHandler {

    var eventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}

File: screen_brightness_ios/ios/screen_brightness_ios/Sources/screen_brightness_ios/StreamHandler/BaseStreamHandler.swift
//
//  BaseEventHandler.swift
//  screen_brightness
//
//  Created by Jack on 25/10/2021.
//

import Foundation
import Flutter

open class BaseStreamHandler: NSObject, FlutterStreamHandler {
    
    var eventSink: FlutterEventSink?
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}
